### DISCUSSION TITLE
Bot "github-actions" not allowed to comment locked PR

### DISCUSSION THREAD

**Discussion by alexbde** [community/community#25230] [Actions]
<p>I experienced following behaviour and didn’t know where I should open an issue.</p>
<p>I’m using an <a href="https://github.com/hashicorp/terraform-github-actions" rel="nofollow noopener">action</a> that runs <code>on PR</code> and comments back with some information at same PR. While PR is unlocked everything works fine. When I’m locking PR only contributors are allowed to take part in the conversation (= creating comments). This behaviour works well for human users but I would have expected technical users like github-actions bot being able to create comments in locked state.</p>
<p>Technical users are configured and maintained by contributors so I understand them as “deputies” and think they should have contributor rights (at least for commenting).</p>
<p>What do you think? If you agree: where should I file an issue?</p>
<p>Example: between locking and unlocking PR in the image below I did same action and expected a comment just like the other ones (<a href="https://github.com/hashicorp/terraform-github-actions/issues/100" rel="nofollow noopener">source</a>).</p>
<p><img src="https://user-images.githubusercontent.com/17784818/68149271-e1ecef00-ff3d-11e9-85c5-ca1997dd166f.png" alt="Image"></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Sorry for the delay response. When you use GITHUB_TOKEN , all of the interactions with the repository are on behalf of the Github-actions bot.</p>
<p>Github-actions bot doesn’t have enough permission to create comments in locked pull request.  </p>
<p>According to the message : You and other members of teams with write access to this repository can still leave comments that others can see.</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181089711-b9ed5df5-e701-4614-94a6-bc5d3d2258af.png" alt="lock pr.png" data-base62-sha1="d81JxIETUu2V0IC3KSfNQ8IXjxy" width="447" height="358"></p>
<p>As a work around, I would suggest you to use your own PAT in the action for adding comment to Pull request. You can store your PAT in secrets and use ${{ secrets.name }} in action.</p>
<p>For Creating and using encrypted secrets, please refer to this document:</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fcreating-and-using-encrypted-secrets&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C93c8edc6add34c9f339a08d768e4c36b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637093204045637186&amp;sdata=i0Tj1%2BqqQnnF%2FpAYQabrWtTnX4pX4%2B7Di3ySx7Zdt7c%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/6e1a600092411542b7a9c64d3fe0d5cdbafc8af2.png" data-download-href="/uploads/short-url/fI11R6IMVsQNyAiMUEpoQrmQGoq.png?dl=1" title="PAT.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089712-88b8b5b2-7233-4430-ad77-7b8d010b7bd0.png" alt="PAT.png" data-base62-sha1="fI11R6IMVsQNyAiMUEpoQrmQGoq" width="690" height="284" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/6e1a600092411542b7a9c64d3fe0d5cdbafc8af2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">PAT.png</span><span class="informations">736×303 5.79 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>In pull request page , you will see that it is you add a comment.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/847a0bcf5d1e74867b8d0f64221aab0ab719d00f.png" data-download-href="/uploads/short-url/iTWvxzjN0kNwlPvg3DZ3UF6fEh9.png?dl=1" title="pull request comment.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089714-a13b4457-941c-4055-8604-215200738077.png" alt="pull request comment.png" data-base62-sha1="iTWvxzjN0kNwlPvg3DZ3UF6fEh9" width="690" height="308" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/847a0bcf5d1e74867b8d0f64221aab0ab719d00f_2_690x308.png, https://aws1.discourse-cdn.com/github/original/1X/847a0bcf5d1e74867b8d0f64221aab0ab719d00f.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/847a0bcf5d1e74867b8d0f64221aab0ab719d00f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/847a0bcf5d1e74867b8d0f64221aab0ab719d00f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">pull request comment.png</span><span class="informations">855×382 18.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jsoref** [comment]
<p>While the bot doesn’t have permission to create a comment in a locked issue, it does have permission to unlock an issue, create a comment, and relock the issue:</p>
<aside class="onebox githubcommit" data-onebox-src="https://github.com/check-spelling/check-spelling/commit/330c0c054decfdeb19416faf46081a2353f59d1d">
  <header class="source">

      <a href="https://github.com/check-spelling/check-spelling/commit/330c0c054decfdeb19416faf46081a2353f59d1d" target="_blank" rel="noopener">github.com/check-spelling/check-spelling</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/check-spelling/check-spelling/commit/330c0c054decfdeb19416faf46081a2353f59d1d" target="_blank" rel="noopener">Handle locked PRs</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2022-01-09" data-time="09:05:16" data-timezone="UTC">09:05AM - 09 Jan 22 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/jsoref" target="_blank" rel="noopener">
          <img alt="jsoref" src="https://user-images.githubusercontent.com/2119212/181089716-48e5cf72-b8d9-473d-a999-a8852db83ca5.png" class="onebox-avatar-inline" width="20" height="20">
          jsoref
        </a>
      </div>

      <div class="lines" title="changed 2 files with 34 additions and 0 deletions">
        <a href="https://github.com/check-spelling/check-spelling/commit/330c0c054decfdeb19416faf46081a2353f59d1d" target="_blank" rel="noopener">
          <span class="added">+34</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
